packages/sanity/src/core/bundles/components/dialog/BundleDetailsDialog.tsx

```diff
@@ -22,69 +24,60 @@ export function BundleDetailsDialog(props: BundleDetailsDialogProps): JSX.Elemen
   const formAction = bundle ? 'edit' : 'create'
   const {t} = useTranslation()
 
-  const [value, setValue] = useState<Partial<BundleDocument>>(() => {
-    if (bundle) {
-      return {
-        slug: bundle.slug,
-        title: bundle.title,
-        description: bundle.description,
-        hue: bundle.hue,
-        icon: bundle.icon,
-        publishedAt: bundle.publishedAt,
-      }
-    }
-
-    return DEFAULT_BUNDLE
+  const [value, setValue] = useState((): FormBundleDocument => {
+    return {
+      _id: bundle?._id || createReleaseId(),
+      _type: 'release',
+      title: bundle?.title,
+      description: bundle?.description,
+      hue: bundle?.hue || 'gray',
+      icon: bundle?.icon || 'cube',
+      publishedAt: bundle?.publishedAt,
+    } as const
   })
   const [isSubmitting, setIsSubmitting] = useState(false)
 
   // TODO MAKE SURE THIS IS HOW WE WANT TO DO THIS
   const {setPerspective} = usePerspective()
 
-  const bundleOperation = useCallback(
-    (formValue: Partial<BundleDocument>) => {
-      if (formAction === 'edit' && bundle?._id) {
-        const updatedBundle: Partial<BundleDocument> = {
-          ...formValue,
-          _id: bundle._id,
-        }
-
-        return updateBundle(updatedBundle)
-      }
-      return createBundle(formValue)
+  const submit = useCallback(
+    (formValue: FormBundleDocument) => {
+      return formAction === 'edit' ? updateBundle(formValue) : createBundle(formValue)
     },
-    [bundle?._id, createBundle, formAction, updateBundle],
+    [createBundle, formAction, updateBundle],
   )
 
+  const slug = useMemo(() => {
+    return value.title ? speakingurl(value.title) : undefined
+  }, [value.title])
+
   const handleOnSubmit = useCallback(
     async (event: FormEvent<HTMLFormElement>) => {
-      if (value.slug) {
-        try {
-          event.preventDefault()
-          setIsSubmitting(true)
+      try {
+        event.preventDefault()
+        setIsSubmitting(true)
 
-          const submitValue = {...value, title: value.title?.trim()}
-          await bundleOperation(submitValue)
-          if (formAction === 'create') {
-            setPerspective(value.slug)
-          }
-        } catch (err) {
-          console.error(err)
-          toast.push({
-            closable: true,
-            status: 'error',
-            title: `Failed to ${formAction} release`,
-          })
-        } finally {
-          setIsSubmitting(false)
-          onSubmit()
+        const submitValue = {...value, title: value.title?.trim()}
+        await submit(submitValue)
+        if (formAction === 'create') {
+          setPerspective(`${value._id!}-${slug}`)
```

Comment:
oohh, damn, great catch! I'm tempted to just remove the slug here for now for the sake of simplicity. Can add it back in a follow-up pr.

Reply:
sounds good! I was thinking the slug could be value later in the setPerspective action instead of passing it like the id
```
  const setPerspective = (bundleId: string | undefined, slug?: string) => {
    if (bundleId === 'drafts') {
      router.navigateStickyParam('perspective', '')
    } else {
      router.navigateStickyParam('perspective', `bundle.${bundleId}-${slug}`)
    }
  }
```

Reply:
That's a great suggestion, and we should add this as a second parameter. I made an attempt at it, but discovered that we have a lot of places already that just use the param as-is without any splitting on `-`, so leaving it without slugifying for now.